6_nodejs/week2/app.js

```diff
@@ -0,0 +1,68 @@
+import express from "express";
+import fs from "fs";
+
+const app = express();
+const port = process.env.PORT || 3000;
+
+app.use(express.json());
+//load all document.json
+const documents = JSON.parse(fs.readFileSync("documents.json", "utf8"));
+
+app.get("/", (req, res) => {
+  res.send("This is a search engine");
+});
+
+// GET /search?id=2
+app.get("/search", (req, res) => {
+  const query = req.query.q;
+  if (!query) {
+    res.json(documents);
+  } else {
+    const results = documents.filter((doc) =>
+      Object.values(doc).some(
+        (value) => typeof value === "string" && value.includes(query)
```

Comment:
This is not a requirement that is mentioned in the homework so it is a nitty pick. value.includes is case sensitive so if I call this endpoint with apple instead of Apple I will get nothing. how about we make this case insensitive. i.e, value.toLowerCase().includes(query)? :)